sys/compat/linuxkpi/common/include/linux/err.h

```diff
@@ -40,12 +40,12 @@
 #define IS_ERR_VALUE(x) unlikely((x) >= (uintptr_t)-MAX_ERRNO)
 
 static inline void *
-ERR_PTR(intptr_t error)
+ERR_PTR(long error)
 {
-	return (void *)error;
+	return (void *)(intptr_t)error;
 }
 
-static inline intptr_t
+static inline long
 PTR_ERR(const void *ptr)
 {
 	return (intptr_t)ptr;
```

Comment:
```suggestion
	return (long)(intptr_t)ptr;
```
to avoid implicit conversions? (I replied to your FreeBSD commit)

Reply:
Hmm, won't the implicit conversion DTRT in both purecap and hybrid for this case, or would this need an explicit ptraddr_t type cast for purecap?  I guess it's like a ptraddr_t cast I added in netgraph.  I might not bother doing it upstream for now.

Reply:
Yeah, it will, and we don't currently warn for it, but we probably should just how there's one for integer truncation.